tembo-operator/src/cloudnativepg/hibernate.rs

```diff
@@ -138,25 +153,25 @@ pub async fn reconcile_cluster_hibernation(cdb: &CoreDB, ctx: &Arc<Context>) ->
         }
     });
 
-    let scheduled_backup_suspend_status = scheduled_backup.spec.suspend.unwrap_or_default();
-    let scheduled_backup_value = cdb.spec.stop;
-    let patch_scheduled_backup_spec = json!({
-        "spec": {
-            "suspend": scheduled_backup_value
-        }
-    });
+    // Update ScheduledBackup if it exists
+    if let Err(action) = update_scheduled_backup(&scheduled_backup, cdb, ctx).await {
+        warn!(
+            "Error updating scheduled backup for {}. Requeuing...",
+            cdb.name_any()
+        );
+        return Err(action);
+    }
 
-    // Check if scheduled_backup_suspend_status=false and cdb.spec.stop=true.  If so then patch the scheduled backup suspend status to true
-    if scheduled_backup_suspend_status != cdb.spec.stop {
-        patch_scheduled_backup_merge(cdb, ctx, patch_scheduled_backup_spec.clone()).await?;
-        info!(
-            "Toggled scheduled backup suspend of {} to '{}'",
-            name, scheduled_backup_value
+    // Patch the Pooler deployment to match the hibernation state
+    if let Err(action) = update_pooler_instances(&pooler, cdb, ctx).await {
```

Comment:
Do we care that this is going to create a potential mismatch between `spec.connectionPooler: enabled` and the amount of requested instances?

Reply:
How could that happen?  If you're talking about the number of postgres instances, then this doesn't touch that.  There can only ever be 1 pooler, or at least that is all we currently support.  

`update_pooler_instances` doesn't even touch the `CoreDB`, so it never will conflict with anything.

Reply:
Maybe I understand?  You're talking about the comment maybe?  If so I just updated the wording. It's patching the `Pooler` cluster resource, not the `Deployment` of the Pooler since that is handed by CNPG.